src/main/java/org/truffleruby/core/module/ModuleOperations.java

```diff
@@ -495,68 +495,78 @@ public static InternalMethod lookupMethod(DynamicObject module, String name, Vis
         return method.getVisibility() == visibility ? method : null;
     }
 
-    public static MethodLookupResult lookupSuperMethod(InternalMethod currentMethod, DynamicObject objectMetaClass) {
+    public static MethodLookupResult lookupSuperMethod(InternalMethod currentMethod, DynamicObject objectMetaClass,
+            DeclarationContext declarationContext) {
+
         assert RubyGuards.isRubyModule(objectMetaClass);
         final String name = currentMethod.getSharedMethodInfo().getName(); // use the original name
         return lookupSuperMethod(
                 currentMethod.getDeclaringModule(),
-                null,
+                new SuperMethodLookup(null, false, null),
                 name,
                 objectMetaClass,
-                currentMethod.getDeclarationContext());
+                declarationContext).getResult();
```

Comment:
I think an additional wrapper should not affect performance, but we can do benchmarks again (if we have thme about `super` execution)